### ISSUE TITLE
Having trouble building on macOS High Sierra (10.13.3)

### ISSUE THREAD

**Issue by malchata** [google/pik#37]
When I attempt to compile using `make -j8` in macOS High Sierra, I receive the following error:

```
image_io.cc:1031:31: error: assigning to 'boolean' from incompatible type 'bool'
  cinfo.do_fancy_upsampling = true;
                              ^~~~
```

Full compiler output: [compiler_output.log](https://github.com/google/pik/files/1693480/compiler_output.log)

Is there anything I'm doing wrong, or is this a legitimate error? Here's my system info (using a Kaby Lake Intel, currently):

Hardware Overview:
  Model Identifier:	MacBookPro14,2
  Processor Name:	Intel Core i5
  Processor Speed:	3.1 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	4 MB
  Memory:	16 GB
  Boot ROM Version:	MBP142.0169.B00
  SMC Version (system):	2.44f1

**Comment by jan-wassenberg** [comment]
Thanks for reporting this. Looks like the High Sierra compiler is more strict.
Can you confirm that `c++ --version` reports LLVM 7.3.0?
We'll update the code to avoid the type mismatch.
In the mean time, you can try defining `HAVE_BOOLEAN=1` and `boolean=bool` before `#include "jpeglib.h"`.

**Comment by jan-wassenberg** [comment]
Should be fixed by 200c78409436fee4426828ef182c08e40e52c25d, please re-open if not :)

**Comment by malchata** [comment]
It works! Thank you so much @jan-wassenberg! You rule!
